### PR TITLE
[JSON]Music library "lastupdated" property

### PR DESCRIPTION
**Pull request by DaveTBlake**
JSON API schema version bumped to v9.3.0 to reflect the (non-breaking) additions to the interface.

- Add "librarylastupdated" property to **AudioLibrary.GetProperties** that returns when the music library was last updated by scanning music files or scraping additional info for artists or albums.
- Add "singlesonly" param to **AudioLibrary.GetSongs**. to fetch only singles.
- Add "lastplayed" field for **AudioLibrary.GetAlbums** and **GetAlbumDetails** previously ommited by accident

Music DB version bumped to v71 for storing when library last updated in versiontagscan table

This will provide better support for JSON consumers trying to sync with updates to the music library.

### CODE REVIEW

**Comment by DaveTBlake** [comment]
Example JSON calls used for testing:
```
{"jsonrpc":"2.0", "id":1, "method":"AudioLibrary.GetProperties",
"params":{"properties":["librarylastupdated"]}}
```

```
{"jsonrpc":"2.0","id":1,"method":"audioLibrary.GetSongs",
"params":{ "properties":["track", "displayartist", "title", "lastplayed"],  
"includesingles": true, "singlesonly": true }}
```

```
{"jsonrpc": "2.0", "id": 1, "method": "AudioLibrary.GetAlbums", 
"params": { "limits": { "start" : 0, "end": 1000 }, "properties": ["lastplayed"] }}
```

**Comment by DaveTBlake** [comment]
These additions test out fine as best I can tell. Looking to merge this tomorrow, unless there are objections, so I can raise my next work with further JSON and music db changes.

File: xbmc/music/MusicDatabase.cpp [hunk 1]
```diff
@@ -5397,6 +5409,18 @@ void CMusicDatabase::SetMusicTagScanVersion(int version /* = 0 */)
...
 
+std::string CMusicDatabase::GetLibraryLastUpdated()
+{
+  return GetSingleValue("SELECT lastscanned FROM versiontagscan LIMIT 1");
+}
+
+void CMusicDatabase::SetLibraryLastUpdated()
+{
+  CDateTime dateUpdated = CDateTime::GetCurrentDateTime();
+  m_pDS->exec(PrepareSQL("UPDATE versiontagscan SET lastscanned = '%s'", dateUpdated.GetAsDBDateTime().c_str()));
```
[the remark this answers is not in the document]
Comment (DaveTBlake) [reply xbmc/music/MusicDatabase.cpp]: No, it is a single row table, but thanks for the interest

**Comment by DaveTBlake** [comment]
@rmrector the primary purpose of  "librarylastupdated" is to enable JSON clients to identify if the library has been rescanned and thus some (maybe all) of song ids have  changed. It is a blunt tool, but better than the current situation where there is no way to know if the previously fetched IDs are now meaningless leaving clients no choice but to repeatedly refetch _all_ the data. Very inefficient when the music library is possibly static for long periods of time.

I don't think that `AudioLibrary.SetSongDetails` changing details (but not id) of a single song should cause this cross library date to be set. A true dufferential sync facility - being able to identify small changes, and what specific items have changed -  for  is something for a another day. Sadly not something I can acheive now, but I do have it in mind.

**Comment by rmrector** [comment]
Sounds reasonable, thanks!

**Pull request merged**